shared/actions/provision.tsx

```diff
@@ -409,9 +404,21 @@ function* startProvisioning(state: Container.TypedState) {
     })
     ProvisioningManager.getSingleton().done('provision call done w/ success')
   } catch (finalError) {
-    ProvisioningManager.getSingleton().done(
-      'provision call done w/ error' + finalError ? finalError.message : ' unknown error'
-    )
+    if (ProvisioningManager.getSingleton() !== manager) {
+      // Another provisioning session has started while this one was active.
+      // This is not desired and is an indication of a problem somewhere else.
+      logger.error(
+        `Provision.startProvisioning error, and ProvisioningManager has changed: ${finalError.message}`
+      )
+      return
+    }
+
+    if (Constants.errorCausedByUsCanceling(finalError) && manager.isCanceled()) {
+      // After cancelling the RPC we are going to get "input canceled" error.
+      return
+    }
+
+    manager.done('provision call done w/ error' + finalError ? finalError.message : ' unknown error')
```

Comment:
below i.e. here? if I move this back to right before the `if (Constants.errorCausedByUsCanceling(finalError) && manager.isCanceled()) {` block for example?

Reply:
Oh i see, else we risk never setting it to done. I guess it should go first in the catch?

Reply:
probably even before the singleton check, yes